client/src/common/components/Members/components/MemberDetails.jsx

```diff
@@ -62,18 +63,20 @@ class MemberDetails extends PureComponent {
       _id: userId
     } = this.props;
 
-    switch (route) {
-      case Routes.COHORT:
-        removeCohortMember(id, userId, isOwner);
-        break;
-      case Routes.LIST:
-        removeListMember(id, userId, isOwner);
-        break;
-      default:
-        break;
-    }
+    const action =
+      route === Routes.COHORT ? removeCohortMember : removeListMember;
+
+    this.setState({ pending: true });
 
-    onClose();
+    action(id, userId, isOwner)
+      .then(() => {
+        this.setState({ pending: false });
+        onClose();
+      })
+      .catch(() => {
+        this.setState({ pending: false });
+        onClose();
```

Comment:
can be replaced with `finally`:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally#Examples